cmd2/cmd2.py

```diff
@@ -130,7 +133,9 @@ def __init__(self, completekey: str = 'tab', stdin=None, stdout=None, *,
                  startup_script: str = '', use_ipython: bool = False,
                  allow_cli_args: bool = True, transcript_files: Optional[List[str]] = None,
                  allow_redirection: bool = True, multiline_commands: Optional[List[str]] = None,
-                 terminators: Optional[List[str]] = None, shortcuts: Optional[Dict[str, str]] = None) -> None:
+                 terminators: Optional[List[str]] = None, shortcuts: Optional[Dict[str, str]] = None,
+                 command_sets: Optional[Iterable[CommandSet]] = None,
+                 auto_load_commands: bool = True) -> None:
```

Comment:
Is there any risk of backwards incompatibility with the default value of `auto_load_commands` being `True`?

I'm thinking we should make the default `False` for safe backwards compatibility.

Reply:
No risk to backwards compatibility here. The option only uses the new decorators and base classes that didn't exist before. I want the default behavior to be that things 'just work' by defining functions and classes using the decorators and base class provided. The extra work of disabling this is for people that have more specific behavior they want to control. For projects that existed prior to this they wouldn't be using the new decorator and base class so they'll be unaffected.